libjcat/jcat-self-test.c

```diff
@@ -1679,151 +1679,465 @@ jcat_rfc6962_func2(void)
+
+	ctv = consistencyTestVectorNew();
+	ctv->snapshot1 = 1;
+	ctv->snapshot2 = 8;
+	g_ptr_array_add(ctv->proof,
+			dh("96a296d224f285c67bee93c30f8a309157f0daa35dc5b87e410b78630a09cfc7", 32));
+	g_ptr_array_add(ctv->proof,
+			dh("5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e", 32));
+	g_ptr_array_add(ctv->proof,
+			dh("6b47aaf29ee3c2af9af889bc1fb9254dabd31177f16232dd6aab035ca39bf6e4", 32));
+	g_ptr_array_add(rv, ctv);
+
+	ctv = consistencyTestVectorNew();
+	ctv->snapshot1 = 6;
+	ctv->snapshot2 = 8;
+	g_ptr_array_add(ctv->proof,
+			dh("0ebc5d3437fbe2db158b9f126a1d118e308181031d0a949f8dededebc558ef6a", 32));
+	g_ptr_array_add(ctv->proof,
+			dh("ca854ea128ed050b41b35ffc1b87b8eb2bde461e9e3b5596ece6b9d5975a0ae0", 32));
+	g_ptr_array_add(ctv->proof,
+			dh("d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7", 32));
+	g_ptr_array_add(rv, ctv);
+
+	ctv = consistencyTestVectorNew();
+	ctv->snapshot1 = 2;
+	ctv->snapshot2 = 5;
+	g_ptr_array_add(ctv->proof,
+			dh("5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e", 32));
+	g_ptr_array_add(ctv->proof,
+			dh("bc1a0643b12e4d2d7c77918f44e0f4f79a838b6cf9ec5b5c283e1f4d88599e6b", 32));
+	g_ptr_array_add(rv, ctv);
+
+	return rv;
+}
+
+static GPtrArray *
+roots(void)
+{
+	GPtrArray *rv = g_ptr_array_new_with_free_func((GDestroyNotify)g_bytes_unref);
+	g_ptr_array_add(rv,
+			dh("6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d", 32));
+	g_ptr_array_add(rv,
+			dh("fac54203e7cc696cf0dfcb42c92a1d9dbaf70ad9e621f4bd8d98662f00e3c125", 32));
+	g_ptr_array_add(rv,
+			dh("aeb6bcfe274b70a14fb067a5e5578264db0fa9b51af5e0ba159158f329e06e77", 32));
+	g_ptr_array_add(rv,
+			dh("d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7", 32));
+	g_ptr_array_add(rv,
+			dh("4e3bbb1f7b478dcfe71fb631631519a3bca12c9aefca1612bfce4c13a86264d4", 32));
+	g_ptr_array_add(rv,
+			dh("76e67dadbcdf1e10e1b74ddc608abd2f98dfb16fbce75277b5232a127f2087ef", 32));
+	g_ptr_array_add(rv,
+			dh("ddb89be403809e325750d3d263cd78929c2942b7942a34b77e122c9594a74c8c", 32));
+	g_ptr_array_add(rv,
+			dh("5dc9da79a70659a9ad559cb701ded9a2ab9d823aad2f4960cfe370eff4604328", 32));
+	return rv;
+}
+
+static GPtrArray *
+leaves(void)
+{
+	GPtrArray *rv = g_ptr_array_new_with_free_func((GDestroyNotify)g_bytes_unref);
+	g_ptr_array_add(rv, dh("", 0));
+	g_ptr_array_add(rv, dh("00", 1));
+	g_ptr_array_add(rv, dh("2021", 2));
+	g_ptr_array_add(rv, dh("3031", 2));
+	g_ptr_array_add(rv, dh("40414243", 4));
+	g_ptr_array_add(rv, dh("5051525354555657", 8));
+	g_ptr_array_add(rv, dh("606162636465666768696a6b6c6d6e6f", 16));
+	return rv;
+}
 
 // inclusionProbe is a parameter set for inclusion proof verification.
-type inclusionProbe struct {
-	gint64 leafIndex
-	gint64 treeSize
+struct inclusionProbe {
+	gint64 leafIndex;
+	gint64 treeSize;
 	GByteArray *root;
 	GByteArray *leafHash;
 	GPtrArray *proof;
 
-	desc string
+	// A string literal for the description. Do not free.
```

Comment:
/* c style comments please */

Reply:
I think I've converted all comments to C style in my change. If there's any remaining that I forgot please let me know.